### PR TITLE
Refs Task #402504 test show page dashboard

### PR DESCRIPTION
**Pull request by nguyenthinhhn**
## Related Tickets

- [#402504](https://dev.sun-asterisk.com/issues/402504)

## What's this PR do ?

- [x] page dashboard

## Performance

- [ ] Resolved n + 1 query
- [ ] Run explain query already
- [ ] Time run rake task : 1000 ms

## Checklist

- [x] It was tested in local success?
- [x] Fill link PR into ticket and the opposite
- [x] Note purpose/reason, solution, scope of influence into ticket

## ENV note

```
```

## Notes
*(Other notes)*

### CODE REVIEW

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr></table>

**Comment by nguyenthinhhn** [comment]
ready

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>failure</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr></table>

File: tests/Feature/DashboardControllerTest.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        $user = factory(User::class)->create(['role' => UserType::USER]);
+
+        $this->actingAs($user);
+        $response = $this->get('/dashboard');
+        $response->assertStatus(200);
+        $response->assertViewHas('countData');
+        $response->assertViewHas('payloadHistory');
+        $response->assertViewHas('messageHistory');
+    }
+    /**
```
Comment (ghost) [inline tests/Feature/DashboardControllerTest.php:67]: xuống 1 line đi e

**Review by ghost — APPROVED** [review]

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr></table>

**Review by tamtt-1031 — APPROVED** [review]

File: tests/Feature/AdminDashboardControllerTest.php [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    * @return void
+    */
+    public function testAdminDashboardFeature()
+    {
+        $user = factory(User::class)->create(['role' => UserType::ADMIN]);
+
+        $this->actingAs($user);
+        $response = $this->get('/admin/dashboard');
+        $response->assertStatus(200);
+        $response->assertViewHas('countData');
```
Comment (ghost) [inline tests/Feature/AdminDashboardControllerTest.php:84]: e viết như này mỗi cover code chứ chưa cover case chưa được e.
e phải test xem mấy data respont về view này có đúng nữa ko mới ổn chơ

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr></table>

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr></table>

**Comment by nguyenthinhhn** [comment]
ready

File: tests/Feature/AdminDashboardControllerTest.php [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    public function testAdminDashboardFeature()
+    {
+        $admin = factory(User::class)->create(['role' => UserType::ADMIN]);
+
+        $chatbot = Bot::create([
+            'name' => 'name',
+            'bot_key' => 'abc',
+            'user_id' => 1
+        ]);
+        $webhook = Webhook::create([
```
Comment (ghost) [inline tests/Feature/AdminDashboardControllerTest.php:34]: mục đích là e tạo 2 webhook với 2 status là disable và enable phải ko, vậy thì e chỉ cần quan tâm đến status của nó, còn những field khác thì trong webhook factory mình đã có fake rồi. nên ở đây e có thể viết ngắn gọn lại như dưới.
```
$webhook1 = factory(Webhook::class)->create(['status' => 0]);
$webhook2 = factory(Webhook::class)->create(['status' =>1]);
```

File: tests/Feature/AdminDashboardControllerTest.php [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+            'bot_id' => $chatbot->id,
+            'name' => 'name',
+            'token' => Str::random(10),
+            'status' => 0,
+            'description' => 'description',
+            'room_id' => '1',
+            'room_name' => 'Room name',
+        ]);
+
+        $payloadHistory = PayloadHistory::create([
```
Comment (ghost) [inline tests/Feature/AdminDashboardControllerTest.php:55]: sao e ko dungf factory vayaj

File: tests/Feature/AdminDashboardControllerTest.php [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+            'log' => 'log'
+        ]);
+
+        $this->actingAs($admin);
+        $response = $this->get('/admin/dashboard');
+        $response->assertStatus(200);
+        
+        $response->assertViewHas('countData');
+        $countDataResponse = $response->getOriginalContent()->getData()['countData'];
+        $this->assertTrue(1 == $countDataResponse['user']);
```
Comment (ghost) [inline tests/Feature/AdminDashboardControllerTest.php:86]: e dùng `assertEqual` đi nhé

File: tests/Feature/AdminDashboardControllerTest.php (hunk 4 above)
Comment (nguyenthinhhn) [reply tests/Feature/AdminDashboardControllerTest.php:55]: dùng factory như này,     return [
        'user_id' => factory(User::class)->create()->id,
        'bot_id' => factory(Bot::class)->create()->id,
        'name' => $faker->name,
        'token' => Str::random(10),
        'status' => 1,
        'description' => $faker->paragraph,
        'room_id' => $faker->ean8,
        'room_name' => $faker->name,
    ];
không những nó tạo ra webhook mà nó còn tạo thêm cácuser và bot khác nữa

File: tests/Feature/AdminDashboardControllerTest.php (hunk 5 above)
Comment (nguyenthinhhn) [reply tests/Feature/AdminDashboardControllerTest.php:86]: dùng cái này cũng kiểm tra được, có sai đâu a

File: tests/Feature/AdminDashboardControllerTest.php (hunk 3 above)
Comment (nguyenthinhhn) [reply tests/Feature/AdminDashboardControllerTest.php:34]: như lệnh của a , dùng fatory nó sẽ tạo thêm các user và bot khác.

Comment (ghost) [reply tests/Feature/AdminDashboardControllerTest.php:34]: e truyền user_id vào nữa là dk, a thiếu

File: tests/Feature/AdminDashboardControllerTest.php (hunk 5 above)
Comment (ghost) [reply tests/Feature/AdminDashboardControllerTest.php:86]: ko sai e, nhưng equal nó phù hợp, ngắn gọn hơn mà e

File: tests/Feature/AdminDashboardControllerTest.php (hunk 3 above)
Comment (nguyenthinhhn) [reply tests/Feature/AdminDashboardControllerTest.php:34]: truyền user_id nó vẫn tạo ra bản mới, vì bên này là admin

File: tests/Feature/AdminDashboardControllerTest.php (hunk 4 above)
Comment (ghost) [reply tests/Feature/AdminDashboardControllerTest.php:55]: cái nào phụ thuộc thì e truyền vào dk mà

File: tests/Feature/AdminDashboardControllerTest.php (hunk 3 above)
Comment (nguyenthinhhn) [reply tests/Feature/AdminDashboardControllerTest.php:34]: e làm như thế từ trc r nhưng k dk

**Review by ghost — APPROVED** [review]

**Comment by phucqd-1305** [comment]
OK

**Pull request merged**